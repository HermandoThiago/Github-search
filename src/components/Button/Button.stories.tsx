import { Meta, Story } from "@storybook/react";
import { IPropsButton, Button } from "./Button";
import { AiOutlineGithub } from "react-icons/ai";

export default {
  title: "Components/Button",
  component: Button,
  args: {
    title: "Click me",
    color: "primary",
  },
} as Meta;

export const Default: Story<IPropsButton> = (args) => <Button {...args} />;

export const Secondary: Story<IPropsButton> = (args) => (
  <Button {...args} color="secondary" />
);

export const Icon: Story<IPropsButton> = (args) => (
  <Button {...args} icon={<AiOutlineGithub />} />
);
