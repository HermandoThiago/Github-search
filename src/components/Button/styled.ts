import styled, { css } from "styled-components";

type color = 'primary' | 'secondary';

interface IButton {
    color: color;
}

export const StyledButton = styled.button<IButton>`
    height: 100%;
    border: 0;
    padding: 10px 20px;
    font-weight: bold;
    font-size: 0.8rem;
    border-radius: 6px;
    cursor: pointer;
    margin-right: 5px;
    display: flex;
    justify-content: center;
    align-items: center;

    :hover{
        transform: translateY(-2px);
    }

    :active{
        transform: translateY(0);
    }

    svg{
        margin-right: 2px;
        font-size: 1.1rem;
    }

    ${(props) => {
        if(props.color === 'primary'){
            return css`
                background-color: #2A2550;
            `
        }else if(props.color === 'secondary'){
            return css`
                background-color: #E04D01;
            `
        }
    }}

`;
