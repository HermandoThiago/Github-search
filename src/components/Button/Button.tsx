import { StyledButton } from './styled';

export interface IPropsButton {
    title: string;
    icon?: React.ReactNode;
    color: 'primary' | 'secondary';
    click?: () => void;
}

export default function Button({ 
    title = 'Button', 
    icon, 
    color = 'primary',
    click
 }: IPropsButton){
    return (
        <StyledButton color={color} onClick={click}>
            {icon}{title}
        </StyledButton>
    )
}