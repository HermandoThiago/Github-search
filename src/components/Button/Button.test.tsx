import renderer from 'react-test-renderer';
import Button from './Button';

describe('Tests Button', () => {
    it('should match snapshot', () => {
        const component = renderer.create(
            <Button 
                title='click me'
                color='primary'
            />
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});
