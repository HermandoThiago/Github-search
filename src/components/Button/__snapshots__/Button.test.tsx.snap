// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Button should match snapshot 1`] = `
<button
  className="sc-bczRLJ ebkUcz"
  color="primary"
>
  click me
</button>
`;
